### PR TITLE
Implement collapsible top app bar plus animate fab action button

### LINKED ISSUE #2040: [UX] Register view enhancement | Round 2
**Describe the enhancement**
|  image   | image    |

**Implementation**

**Priority 1**
- [x] Update the spacing between the HH member and the ID to match the design
- [x] Header on scroll 
    - [x] When the user scrolls up, the header should contain the HH name on the Household profile. 
    - [x] When the user scrolls down, the header should expand back to show the HH name and Community
    - [ ] both headers on scroll have a dropdown shadow
  
 **Priority 2**
- [ ] Update the following items on the patient register component
    - [x] Update the patient name color to match the designs
    - [ ] The HH linked to the patient is not included
    - [ ] Remove the vertical dividers that are not part of the designs
    - [ ] Update the margin left and right to match the designs
- [ ] Last HH member list 
    - [ ] The container of a HH member has a top-bottom padding of 24px
    - [ ] The last HH member in the members' list should always be above the Add member floating button as shown in the attached design. This is to allow the user to be able to click any actions related to patient

**Priority 3**
- [ ] Pagination
    - [x] The pagination row should be part of the list and not always on the bottom of the screen. 
    - [x] Pagination should show when the user scrolls to the end of the screen
- [ ] Floating button on scroll
   - [x] When the user scrolls up, the floating button should collapse into an only icon
   - [x] When the user scrolls down, the floating button should return to the full button

### PR DESCRIPTION
**Pull request by ellykits**
**IMPORTANT: Where possible all PRs must be linked to a Github issue**

Fixes #2040 

**Checklist**
- [ ] I have written **Unit tests** for any new feature(s) and edge cases for bug fixes
- [ ] I have added any strings visible on UI components to the `strings.xml` file
- [ ] I have updated the  [CHANGELOG.md](./CHANGELOG.md) file for any notable changes to the codebase
- [ ] I have run `./gradlew spotlessApply` and `./gradlew spotlessCheck` to check my code follows the project's style guide
- [ ] I have built and run the fhircore app to verify my change fixes the issue and/or does not break the app

### CODE REVIEW

File: android/engine/src/main/java/org/smartregister/fhircore/engine/ui/theme/Colors.kt [hunk 1]
```diff
@@ -39,7 +39,6 @@ val SideMenuBottomItemDarkColor = Color(0xFF404040)
 val AppTitleColor = Color(0xFF929496)
 val StatusTextColor = Color(0xFF6F7274)
 val PersonalDataBackgroundColor = Color(0xFFF5F5F5)
-val ProfileBackgroundColor = Color(0xFFF2F4F7)
```
Comment (dubdabasoduba) [inline android/engine/src/main/java/org/smartregister/fhircore/engine/ui/theme/Colors.kt:42]: Did you mean to remove this?

Comment (dubdabasoduba) [reply android/engine/src/main/java/org/smartregister/fhircore/engine/ui/theme/Colors.kt:42]: NVM, I answered myself.

File: android/quest/src/main/java/org/smartregister/fhircore/quest/ui/shared/components/CardView.kt [hunk 2]
```diff
@@ -56,7 +57,7 @@ fun CardView(
         Spacer(modifier = modifier.height(8.dp))
         CompoundText(
           modifier = modifier.wrapContentWidth(Alignment.Start),
-          compoundTextProperties = viewProperties.header!!,
+          compoundTextProperties = viewProperties.header!!.copy(textCase = TextCase.UPPER_CASE),
```
Comment (dubdabasoduba) [inline android/quest/src/main/java/org/smartregister/fhircore/quest/ui/shared/components/CardView.kt:60]: Nice!

File: android/quest/src/main/java/org/smartregister/fhircore/quest/ui/shared/components/CompoundText.kt [hunk 3]
```diff
@@ -137,9 +147,21 @@ private fun CompoundTextPart(
...
+  val interpolatedText = text.interpolate(resourceData.computedValuesMap)
   Text(
-    text = text.interpolate(resourceData.computedValuesMap).removeExtraWhiteSpaces(),
-    color = textColor?.interpolate(resourceData.computedValuesMap)?.parseColor() ?: DefaultColor,
+    text =
+      when (textCase) {
+        TextCase.UPPER_CASE -> interpolatedText.uppercase()
+        TextCase.LOWER_CASE -> interpolatedText.lowercase()
+        TextCase.CAMEL_CASE -> interpolatedText.camelCase()
+        null -> interpolatedText
```
Comment (dubdabasoduba) [inline android/quest/src/main/java/org/smartregister/fhircore/quest/ui/shared/components/CompoundText.kt:157]: Nice! addition

**Review by dubdabasoduba — APPROVED** [review]

**Comment by dubdabasoduba** [comment]
I haven't tested this on a device but my initial pass looks good.

File: android/engine/build.gradle [hunk 4]
```diff
@@ -168,8 +168,9 @@ dependencies {
     api("androidx.navigation:navigation-testing:$navigationVersion")
     api("androidx.compose.material:material-icons-extended:1.3.1")
     api("androidx.compose.material:material-icons-core:1.3.1")
+    api("androidx.compose.material3:material3:1.0.1")
     api("androidx.compose.foundation:foundation:1.3.1")
-    api("androidx.compose.material:material:1.3.1")
+//    api("androidx.compose.material:material:1.3.1")
```
Comment (ndegwamartin) [inline android/engine/build.gradle:173]: Should we remove this entirely?

File: android/engine/src/main/java/org/smartregister/fhircore/engine/configuration/view/CompoundTextProperties.kt [hunk 5]
```diff
@@ -57,3 +60,9 @@ enum class TextFontWeight(val fontWeight: FontWeight) {
   EXTRA_LIGHT(FontWeight.ExtraLight),
   SEMI_BOLD(FontWeight.SemiBold),
 }
+
+enum class TextCase {
+  UPPER_CASE,
+  LOWER_CASE,
+  CAMEL_CASE,
```
Comment (ndegwamartin) [inline android/engine/src/main/java/org/smartregister/fhircore/engine/configuration/view/CompoundTextProperties.kt:67]: **Pascal Case** is quite common as well, do we add it?

**Comment by ellykits** [comment]
Yup

File: android/engine/src/main/java/org/smartregister/fhircore/engine/configuration/view/CompoundTextProperties.kt (hunk 5 above)
Comment (ellykits) [reply android/engine/src/main/java/org/smartregister/fhircore/engine/configuration/view/CompoundTextProperties.kt:67]: Sure

Comment (ellykits) [reply android/engine/src/main/java/org/smartregister/fhircore/engine/configuration/view/CompoundTextProperties.kt:67]: We can include it when we have a use case for it.

**Pull request merged**